blast/ncbi/doc/man/findspl.1
.TH FINDSPL 1 2001-10-05 NCBI "NCBI Tools User's Manual"
.SH NAME
findspl \- find protein splice sites
.SH SYNOPSIS
.B findspl
[\|\fB\-\fP\|]
[\|\fB\-g\fP\ \fIN\fP\|]
[\|\fB\-i\fP\ \fIfilename\fP\|]
[\|\fB\-o\fP\ \fIfilename\fP\|]
.SH DESCRIPTION
\fBfindspl\fP is a simple tool to find protein splice sites.
.SH OPTIONS
A summary of options is included below.
.TP
\fB\-\fP
Print usage message
.TP
\fB\-g\fP\ \fIN\fP
GI number of protein
.TP
\fB\-i\fP\ \fIfilename\fP
Input file containing protein GIs (default is stdin)
.TP
\fB\-o\fP\ \fIfilename\fP
Output file (default is stdout)
.SH AUTHOR
The National Center for Biotechnology Information.


QuickTimeStreamingServer/ReleaseNotes.txt

                                 RELEASE NOTES
                             Darwin Streaming Server 

These release notes are for the current development version of Darwin
Streaming Server.

------------------------------------------------------------------------
Date: 8/7/2001
Version: 3.0.1+development [v349]
cvs tag: qtssServer-349

*   Updated build platforms: Solaris 8, Redhat Linux 7.1, FreeBSD 4.3, 
    Mac OS X 10.1
*   Retransmit/overbuffering (Skip Protection) algorithms tuned
    substantially
*   "reliable_udp" attribute defaults to "enabled" 
*   Various WebAdmin fixes for Windows platform

------------------------------------------------------------------------
Date: 8/9/2001
Version: 3.0.1+development [v350]
cvs tag: qtssServer-350

*   Minor bug fixes.

------------------------------------------------------------------------
Date: 8/14/2001
Version: 3.0.1+development [v351]
cvs tag: qtssServer-351

*   QTSSFileModule - enabled caching based on preferences (shared and
    private).
*   Add buffer delay to on demand movies.
*   QTFile - code cleanup.
*   Changed some paths for the installer.
*   Added support for MP3Broadcaster on Windows.

------------------------------------------------------------------------
Date: 8/16/2001
Version: 3.0.1+development [v352]
cvs tag: qtssServer-352

*   Added MP3StreamingModule to all builds.
*   Fixed gdb script to execute QTSS from build dir on MacOSX.

------------------------------------------------------------------------
Date: 8/19/2001
Version: 3.0.1+development [v353]
cvs tag: qtssServer-353

*   Added:
        add_seconds_to_client_buffer_delay -- Increases client buffer
                                delay before playing movie.
        record_movie_file_sdp -- Records a .sdp file for a .mov file.
                                Appears as moviename.mov.sdp
        enable_movie_file_sdp -- Plays a .mov.sdp file for a .mov movie.
*   Fixed server to play .mov.sdp files.
*   QTSSFileModule enables both private and shared movie cache buffers.

------------------------------------------------------------------------
Date: 8/23/2001
Version: 3.0.1+development [v354]
cvs tag: qtssServer-354

*   Added prefs for MP3StreamingModule.
*   Default Reliable UDP dir is / on UNIX and \ on NT
*   Use Enable Reliable UDP pref to disable Reliable UDP
*   Included new Web Admin
*   Changed relay preferences to XML format.
        Old relay config files will not work.

------------------------------------------------------------------------
Date: 08/28/2001
Version: 3.0.1+development [v355]
cvs tag: qtssServer-355

*	Added preference:
	 RTSP_error_message to enable debugging error strings returned in 
	 the content body of RTSP error responses.
	
*	Changed RTSP code to:
	1) require a Cseq field for OPTIONS requests
	2) allow missing Range headers in a PLAY request (defaults to entire movie)
	3) add content length header to Error responses containing error messages
	4) add SDP o= and t= fields to on-demand movie Describe responses

*      Changed Relay to relay incoming ANNOUNCEd broadcasts
       1) All ANNOUNCEd broadcasts, or
       2) All broadcasts from an IP address
       3) A broadcast URL from an IP address

       (a) Can still configure the relay to relay UDP streams
       (b) Can pull an SDP or movie from source and relay it

*      Changed Relay to ANNOUNCE the SDPs to its destinations
       1) for ANNOUNCED sources
       2) for sources that are pulled from the source server

*      Changed Relay config file to be in XML format
       1) relayconfig.xml-Sample will be installed instead of old streamingrelay.conf

*	Added General Settings and Change Password to UI
	1) Require existing password to change to new
	2) Require the user to enter the same password twice
	3) Change the auth cookie so the user won't have to re-authenticate

-------------------------------------------------------------------------------------------------
Date: 8/31/2001
Version: 3.0.1+development [v356]
cvs tag: qtssServer-356

*	 Admin Module change
	 1) Added an admin module pref called "AdministratorGroup" with
	 a default value of "admin"
	 2) Changed Admin Module to allow all users that belong to the
	 AdministratorGroup, instead of allowing only one user called
	 "streamingadmin".
	 3) A default qtgroups gets installed from now. The default qtusers
	 file has user 'streamingadmin' with password 'default'. The default
	 qtgroups file has the group 'admin' with the user 'streamingadmin'.
	 4) The login name for the web admin can be changed by adding the user
	 to the qtusers file and adding the user to the 'admin' group in the
	 qtgroups file.

*	Added "RTSP_debug_printfs" server pref to show the incoming and outgoing RTSP.
	Fixed Session header response to remove ";timeout=" field if the value is not defined.
	Fixed .mov.sdp Describe response to include the Last-Modified and Cache-Control headers.
	Fixed RTSP Session header parser to also stop on a ';'
	Changes shared file buffer size to default to maximum size if the bit-rate is 0 and no fixed 
	buffer size is defined.

-------------------------------------------------------------------------------------------------
Date: 9/04/2001
Version: 3.0.1+development [v357]
cvs tag: qtssServer-357

*	more implementation code for Web admin added.
*	New XML parser for reading prefs.
*	full implementation of MP3 streaming module is included.

-------------------------------------------------------------------------------------------------
Date: 9/06/2001
Version: 3.0.1+development [v359]
cvs tag: qtssServer-359

*	Relay that pulls source from the same server can be setup now
	[rtsp_source SOURCE with own IP address as the source_addr]
   
-------------------------------------------------------------------------------------------------
Date: 9/10/2001
Version: 3.0.1+development [v361]
cvs tag: qtssServer-361

*	UI: play and stop buttons in playlist screen now start/stop playlists immediately

blast/ncbi/demo/findspl.c
/*   findspl.c
* ===========================================================================
*
*                            PUBLIC DOMAIN NOTICE                          
*               National Center for Biotechnology Information
*                                                                          
*  This software/database is a "United States Government Work" under the   
*  terms of the United States Copyright Act.  It was written as part of    
*  the author's official duties as a United States Government employee and 
*  thus cannot be copyrighted.  This software/database is freely available 
*  to the public for use. The National Library of Medicine and the U.S.    
*  Government have not placed any restriction on its use or reproduction.  
*                                                                          
*  Although all reasonable efforts have been taken to ensure the accuracy  
*  and reliability of the software and data, the NLM and the U.S.          
*  Government do not and cannot warrant the performance or results that    
*  may be obtained by using this software or data. The NLM and the U.S.    
*  Government disclaim all warranties, express or implied, including       
*  warranties of performance, merchantability or fitness for any particular
*  purpose.                                                                
*                                                                          
*  Please cite the author in any work or product based on this material.   
*
* ===========================================================================
*
* File Name:  findspl.c
*
* Author:  Epstein, Weisemann
*
* Version Creation Date:   1/4/94
*
* $Revision: 6.2 $
*
* File Description: 
*       Find protein splice sites
*
* Modifications:  
* --------------------------------------------------------------------------
* Date     Name        Description of modification
* -------  ----------  -----------------------------------------------------
*
* $Log: findspl.c,v $
* Revision 6.2  1998/08/26 18:41:09  kans
* fixed remaining -v -fd warning
*
* Revision 6.1  1998/08/24 20:43:43  kans
* fixed -v -fd warnings
*
* Revision 6.0  1997/08/25 18:20:00  madden
* Revision changed to 6.0
*
* Revision 1.2  1996/06/24 18:37:50  epstein
* move splicing functionality to sequtil.c
*
 * Revision 1.1  1996/06/19  15:23:21  epstein
 * Initial revision
 *
*
* ==========================================================================
*/

#include <accentr.h>
#include <sequtil.h>

#define NUMARGS 3
Args myargs[NUMARGS] = {
        {"Gi number of protein","0","1","99999999",FALSE,'g',ARG_INT,0.0,0,NULL},
	{"Input file containing protein GIs","stdin","NULL","NULL",FALSE,'i',ARG_FILE_IN,0.0,0,NULL},
	{"Output file","stdout","NULL","NULL",FALSE,'o',ARG_FILE_OUT,0.0,0,NULL}};


static void FindAndPrint(FILE *ofp, Int4 gi)
{
	SeqLocPtr slp, current;
	SeqEntryPtr sep;

        if ((sep = EntrezSeqEntryGet(gi, 0)) != NULL)
        {
            fprintf (ofp, "%ld", (long) gi);
	    slp = FindSpliceSites(sep, /* find on protein */ TRUE);
            for (current = SeqLocFindNext(slp, NULL); current != NULL;
                 current = SeqLocFindNext(slp, current))
	    {
		fprintf (ofp, " %ld", (long) SeqLocStart(current) + 1);
            }

            fprintf (ofp, "\n");
            SeqEntryFree(sep);
            SeqLocFree(slp);
        }
}

Int2 Main(void)
{
	long longgi;
	char line[9];
	Int4 gi;
	FILE *ofp;
	FILE *ifp = NULL;

	if ( !GetArgs("findspl", NUMARGS, myargs) ) return 1;

	gi = myargs[0].intvalue;
	if (gi <= 0)
	{
		if ((ifp = FileOpen(myargs[1].strvalue, "r")) == NULL)
		{
			Message(MSG_FATAL, "Unable to open input file %s",
		                myargs[1].strvalue);
			return 1;
		}
	}

	if ((ofp = FileOpen(myargs[2].strvalue, "w")) == NULL)
	{
		Message(MSG_FATAL, "Unable to open output file %s",
	                myargs[2].strvalue);
		return 1;
	}

	if (! EntrezInit("findspl", FALSE, NULL))
	{
		Message(MSG_FATAL, "Unable to open Entrez database");
		FileClose(ofp);
		return 1;
	}

	if (gi > 0)
	{
            FindAndPrint(ofp, gi);
	} else {
	    while (fgets(line,sizeof(line),ifp) != NULL) {
	        sscanf(line, "%ld", &longgi);
	        gi = (Int4) longgi;
                FindAndPrint(ofp, gi);
	    }
	    FileClose(ifp);
	}

	FileClose(ofp);
	EntrezFini();

	return 0;
}
